File: src/updaters/lazygit.ts
import { invoke } from "@tauri-apps/api/core";
import type { UpdateResult } from "../types/updaters.ts";
import type { UpdaterContext } from "./registry.ts";

export async function runLazygitUpdater(ctx: UpdaterContext): Promise<UpdateResult> {
    const { themeKey, collectionKey, appConfig } = ctx;
    const { config_path, themes_path } = appConfig;

    if (!themes_path) {
        return {
            app: "lazygit",
            status: "error",
            message: "Missing themes_path",
        };
    }

    const sourcePath = `${themes_path}/${collectionKey}/${themeKey}.yml`;

    try {
        await invoke("patch_yaml_file", {
            targetPath: config_path,
            sourcePath: sourcePath,
        });

        // No reload needed — lazygit reads config on launch
        return { app: "lazygit", status: "done" };
    } catch (error) {
        const message = error instanceof Error ? error.message : String(error);
        console.warn("[lazygit updater]", error);
        return { app: "lazygit", status: "error", message };
    }
}

File: src-tauri/src/updaters/lazygit.rs
use crate::config::types::AppConfig;

use super::file_ops;
use super::{UpdateContext, UpdateResult};

pub fn update(app_str: &str, app_config: &AppConfig, ctx: &UpdateContext) -> UpdateResult {
    let themes_path = match &app_config.themes_path {
        Some(tp) => tp,
        None => return UpdateResult::error(app_str, "Missing themes_path"),
    };

    let source_path = format!(
        "{}/{}/{}.yml",
        themes_path, ctx.collection_key, ctx.theme_key
    );

    match file_ops::yaml::patch_yaml_file(app_config.config_path.clone(), source_path) {
        Ok(()) => UpdateResult::done(app_str),
        Err(e) => UpdateResult::error(app_str, e),
    }
}
